homeassistant/components/pi_hole/const.py

```diff
@@ -91,3 +93,50 @@ class PiHoleSensorEntityDescription(SensorEntityDescription):
         icon="mdi:domain",
     ),
 )
+
+
+@dataclass
+class PiHoleBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes PiHole binary sensor entity."""
+
+    icon: str = "mdi:pi-hole"
+    version_current: str = ""
+    version_latest: str = ""
+    version_update: str = ""
+
+
+BINARY_SENSOR_TYPES: tuple[PiHoleBinarySensorEntityDescription, ...] = (
+    PiHoleBinarySensorEntityDescription(
+        key="core_update_available",
+        name="Core Update Available",
+        icon="mdi:update",
```

Comment:
Entities that have a device class set normally don't need to define a custom icon. The frontend already has a custom icon set for each device class.
```suggestion
```